rootfs/etc/confd/templates/builder

```diff
@@ -64,17 +64,30 @@ TMP_DIR=$(mktemp -d -p $BUILD_DIR)
 
 cd $REPO_DIR
 # use Procfile if provided, otherwise try default process types from ./release
+
+git archive --format=tar.gz ${GIT_SHA} > ${APP_NAME}.tar.gz
+
+tar -xzf ${APP_NAME}.tar.gz -C $TMP_DIR/
+
 USING_DOCKERFILE=true
-if [ -f Procfile ]; then
-    PROCFILE=$(cat Procfile | yaml2json-procfile)
+if [ -f $TMP_DIR/Procfile ]; then
+    PROCFILE=$(cat $TMP_DIR/Procfile | yaml2json-procfile)
     USING_DOCKERFILE=false
-    POD="slugbuilder"
 else
     PROCFILE="{}"
 fi
+# if [ -f Procfile ]; then
+#     PROCFILE=$(cat Procfile | yaml2json-procfile)
+#     USING_DOCKERFILE=false
+#     POD="slugbuilder"
+# else
+#     PROCFILE="{}"
+# fi
+echo "procfile" $PROCFILE
+echo "dockerfile" $USING_DOCKERFILE
 
 if [[ ! -f /var/run/secrets/object/store/access-key-id  ]]; then
-  if [ -f Dockerfile ]; then
+  if $USING_DOCKERFILE ; then
```

Comment:
I did a ls of files inside repo direcotry and it didn't show up any files related to example-go, but when I did archive and untar the files are there. You can play with smothiki/builder:3.1.0 to observe this behavior. reason for doing a git archive and then untar the files in a temp direcotry. 
This might be issue with previous builder as well but its passing because we have procfile check from generated slug.